### PR TITLE
chore(server): add inventory management for parts and materials

### PR DESCRIPTION
**Pull request by Mackbellemore**
#

## Changes this PR introduces

- Creates default material lists required to build certain types of parts
- Validates stocks for material based on the above list

## How to test this PR

- create POST requests and PATCH requests for parts, note that now you need to pass a valid type to the part for a successful POST Request as well as having the needed materials in the DB. Valid types are:

  'handle_bar',
  'wheels',
  'chain',
  'frame',
  'pedal',
  'brakes',
  'seat',
  'fork',

## Notes for reviewers

- this is a note

## Checklist

- [x] Have I renamed the title of this pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)?
- [x] I will squash and merge when my PR is approved.
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-npzdz3 Link bikes, parts and materials](https://app.clickup.com/t/npzdz3)

File: server/tests/unit/PartController.test.ts [hunk 1]
```diff
@@ -18,13 +21,14 @@ const mockPart = {
   name: 'wheel',
   quality: 'high-quality',
   description: 'Circular frame of hard material that is solid',
-  type: 'round',
```
Comment (gordonpn) [inline server/tests/unit/PartController.test.ts:21]: lmfao

File: server/app/models/PartModel.ts [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
   name: { type: Schema.Types.String, required: true, unique: true, immutable: true },
   quality: { type: Schema.Types.String, required: true },
   description: { type: Schema.Types.String, required: false },
-  type: { type: Schema.Types.String, required: true },
+  type: { type: Schema.Types.String, required: true, immutable: true },
   color: { type: Schema.Types.String, required: false },
   finish: { type: Schema.Types.String, required: false },
   grade: { type: Schema.Types.String, required: false },
   detail: { type: Schema.Types.String, required: false },
+  stock: { type: Schema.Types.Number, required: true },
```
Comment (gordonpn) [inline server/app/models/PartModel.ts:25]: What happens if we `post` with a negative part stock amount?

Comment (Mackbellemore) [reply server/app/models/PartModel.ts:25]: Entire app explodes and zeet kicks us off the platform

Comment (gordonpn) [reply server/app/models/PartModel.ts:25]: that's what i thought

Comment (Mackbellemore) [reply server/app/models/PartModel.ts:25]: kidding, validation happens at the controller level now. I was avoiding re writing validation in 2 places

Comment (Mackbellemore) [reply server/app/models/PartModel.ts:25]: but we could have double validation here

Comment (gordonpn) [reply server/app/models/PartModel.ts:25]: Ahhh I see it now, I thought I saw it. Didn't know we had 2 schema thingies

**Review by gordonpn — APPROVED** [review]
🧱

**Review by sujvn — APPROVED** [review]

**Pull request merged**